migrate shell-ui and metalk8s-ui to rspack

MonPote opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

MonPote committed:
update package to be ready for node 20

bert-e commented:
# Hello monpote,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/metalk8s/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @aprucolimartins

* @teddyandrieux

* @eg-ayoub

* @ezekiel-alexrod

* @thomasdanan

* @xaf-scality

* @m4nch0t

* @chengyanjin

* @jbwatenbergscality

MonPote committed:
install and build with rspack

MonPote committed:
install and build with rspack metalk8s-ui

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @eg-ayoub

* @jbwatenbergscality

* @m4nch0t

* @chengyanjin

* @teddyandrieux

* @aprucolimartins

* @ezekiel-alexrod

* @xaf-scality

* @thomasdanan

MonPote committed:
shell is running alone

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @ezekiel-alexrod

* @thomasdanan

* @jbwatenbergscality

* @xaf-scality

* @m4nch0t

* @aprucolimartins

* @eg-ayoub

* @chengyanjin

* @teddyandrieux

MonPote committed:
add module federation enhanced to shell

MonPote committed:
add module federation enhanced to metalk8s

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @m4nch0t

* @xaf-scality

* @jbwatenbergscality

* @aprucolimartins

* @eg-ayoub

* @chengyanjin

* @teddyandrieux

* @thomasdanan

* @ezekiel-alexrod

MonPote committed:
SAVE TMPModuleFederation.tsx

MonPote committed:
improve fetching of manifest

MonPote committed:
remove legacy files

MonPote committed:
update shell-ui html template to use mf-manifest instead of remoteEntry

MonPote committed:
update module federation to 1.3.0

MonPote committed:
update rspack config and change manifest url

MonPote committed:
add federated type in the gitignore

MonPote committed:
remove legacy files

MonPote committed:
some cleaning rspack.config.ts

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @thomasdanan

* @jbwatenbergscality

* @eg-ayoub

* @teddyandrieux

* @m4nch0t

* @ezekiel-alexrod

* @xaf-scality

* @aprucolimartins

MonPote committed:
fix jest tests

MonPote committed:
fix jest tests

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @xaf-scality

* @teddyandrieux

* @aprucolimartins

* @ezekiel-alexrod

* @jbwatenbergscality

* @m4nch0t

* @thomasdanan

* @chengyanjin

* @eg-ayoub

MonPote committed:
fix docker build

MonPote committed:
update Dockerfile for rspack

MonPote committed:
remove unused ts expected error

MonPote committed:
update python script for shell-ui

MonPote committed:
add rsdoctor

MonPote committed:
format index-template

MonPote committed:
remove flowconfig

MonPote committed:
fix dev mode

MonPote committed:
update core-ui with the latest icons

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @ezekiel-alexrod

* @aprucolimartins

* @xaf-scality

* @thomasdanan

* @chengyanjin

* @m4nch0t

* @eg-ayoub

* @jbwatenbergscality

* @teddyandrieux

MonPote committed:
update build file paths

MonPote committed:
rm webpack from shell-ui

MonPote committed:
try to remove shell-ui-docker-entrypoint.sh

MonPote committed:
remove not used shared packages

JBWatenbergScality committed:
shell-ui: Introduce window API level to shell-ui

JBWatenbergScality committed:
Update types

JBWatenbergScality committed:
Remove check on shell-ui remote entry

JBWatenbergScality committed:
Add getToken method to useAuth

MonPote committed:
cleanup metalk8s rspack config

MonPote committed:
sync package-lock.json for metalk8s

JBWatenbergScality committed:
Performance improvements, has to be bundled with a new version of core-ui

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @eg-ayoub

* @chengyanjin

* @thomasdanan

* @jbwatenbergscality

* @teddyandrieux

* @xaf-scality

* @ezekiel-alexrod

* @m4nch0t

* @aprucolimartins

MonPote committed:
fix useConfig error on test

MonPote committed:
chore: cleaning DashboardServices.test.tsx

MonPote committed:
improve shared types

MonPote committed:
fix legacy component

MonPote committed:
fix route type

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @aprucolimartins

* @thomasdanan

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod

* @teddyandrieux

* @chengyanjin

* @m4nch0t

* @xaf-scality

MonPote committed:
fix build type in ci

MonPote committed:
update types

MonPote committed:
fix types

MonPote committed:
remove webpack from metalk8s-ui

MonPote committed:
bump node to node20 on build.yaml

MonPote committed:
remove no optionnal binding

MonPote committed:
add mf-types buildchain build

bert-e commented:
# Conflict




There is a conflict between your branch `feature/migrate-to-rspack` and the
destination branch `development/128.0`.

Please resolve the conflict on **the feature branch** (`feature/migrate-to-rspack`).

```sh
git fetch && \
git checkout origin/feature/migrate-to-rspack && \
git merge origin/development/128.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/migrate-to-rspack
```

MonPote committed:
add mf-manifest in metalk8s-ui ingress

MonPote committed:
try

JeanMarcMilletScality committed:
wait the loading of share dependencies before loading micro app

JeanMarcMilletScality committed:
update module federation with types

JeanMarcMilletScality committed:
update module federation

MonPote committed:
add module federation as deps to check

MonPote committed:
add react-router-dom loadshare deps

MonPote committed:
fix GlobalHealthBar import

GlobalHealthBar import cause issue when we laod metalk8s as a
microapp. It's because the chunk are not in the same file as
SyncedCursorCharts.

MonPote committed:
update mf-types

MonPote committed:
remove the need to load all microapp for NavbarUpdaterComponents

MonPote committed:
update dockerfile node version

MonPote committed:
update core-ui

JBWatenbergScality committed:
Fix firefox routing issue

MonPote committed:
fix redux store type

MonPote committed:
update mf-types

MonPote committed:
Merge remote-tracking branch 'origin/development/128.0' into HEAD

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @m4nch0t

* @chengyanjin

* @eg-ayoub

* @xaf-scality

* @teddyandrieux

* @ezekiel-alexrod

* @aprucolimartins

* @thomasdanan

* @jbwatenbergscality

MonPote committed:
fix unit test

MonPote committed:
run rsbuild instead of build to avoid the type checking

We should invest some time to fix these type issue.
We most probably need to change some of our React.ReactNode to
JSX.Element.

Most of type errors are related to this.

JBWatenbergScality committed:
Remove useless useEffect

JBWatenbergScality committed:
Fix end2end tests

MonPote commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @m4nch0t

* @chengyanjin

* @eg-ayoub

* @xaf-scality

* @teddyandrieux

* @ezekiel-alexrod

* @aprucolimartins

* @thomasdanan

* @jbwatenbergscality










*The following options are set:* **approve**

JBWatenbergScality committed:
Fix end2end tests

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/128.0`



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/126.0`
* `development/127.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/128.0`



The following branches have **NOT** changed:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/126.0`
* `development/127.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye monpote.
